### PR TITLE
Add JSON logging

### PR DESCRIPTION
**Pull request by Loonride**
Clubhouse story: https://app.clubhouse.io/globus/story/8973/jsonify-logs-from-funcx-webservice

This logs all current logs now in JSON format as well. 2 key points now have additional log info - when task is submitted, and when task gets queried by user and is complete (The fields that are added are `user_id`, `task_id`, `task_group_id`, `endpoint_id`, etc.)

~A potential issue still remains where the original log still appears along with the JSON log~ (EDIT: this has been resolved)

task_query_logs

### CODE REVIEW

File: funcx_web_service/routes/funcx.py
Comment (ryanchard) [inline funcx_web_service/routes/funcx.py:159]: Could we add function_id too?

Comment (Loonride) [reply funcx_web_service/routes/funcx.py:159]: Yup

Comment (yadudoc) [reply funcx_web_service/routes/funcx.py:159]: Having `function_id` and `container_id` would be useful.

**Comment by yadudoc** [comment]
Hi @Loonride , are you adding `task_transition=Bool` flag to make it easy to search for the transition messages? I'm a bit unclear on what that does. Have you considered putting the new state in the message instead? Right now it looks like the state transition info itself is in human readable form and not quite machine readable.

**Comment by Loonride** [comment]
> Hi @Loonride , are you adding `task_transition=Bool` flag to make it easy to search for the transition messages? I'm a bit unclear on what that does. Have you considered putting the new state in the message instead? Right now it looks like the state transition info itself is in human readable form and not quite machine readable.

Sure, that makes sense. I figured we should also have some boolean specifying it as a `task_transition` so that we can easily filter for all the JSON messages that only have this set to true. Does that make sense?

**Comment by Loonride** [comment]
So far I've added 6 key places where a task is logged. I'm not sure exactly what to call them, need suggestions. I think it doesn't exactly make sense to follow the task states that we already have because, for example, we don't care on the logging side if a task Succeeded/Failed, we just care if it completed. WebSocket service also adds some complexity, since a user could either use the old query system or use the websocket service.
- `received` - web service got task from user
- `complete` - web service was queried by user for task and task was complete (succeeded or failed)
- `forwarded` - forwarder forwarded task to endpoint
- `result-enqueued` - forwarder enqueued received result on rabbitmq
- `result-dequeued` - websocket service dequeued result from rabbitmq because a user connected as a client and queried for the corresponding task group
- `result-sent` - websocket service sent result back to connected user

**Comment by yadudoc** [comment]
Following our chat, here's a quick summary of potential changes we might want here.
* We ideally should distinguish tasks getting fetched by the user vs being sent by websockets.
* We can do this by tagging the state transition as `user_fetched` vs `dispatched_to_user`
* I think it's clearer to call `forwarded` as `dispatched_to_endpoint`

File: funcx_web_service/__init__.py
Comment (Loonride) [inline funcx_web_service/__init__.py:25]: This removes the flask default handler

Comment (yadudoc) [inline funcx_web_service/__init__.py:25]: Add a comment here to clarify why we are removing this.

File: funcx_web_service/models/tasks.py
Comment (Loonride) [inline funcx_web_service/models/tasks.py:117]: Added these fields to Task

**Review by yadudoc — APPROVED** [review]
These look good to go in! Let's deploy to dev once all the hosted bits are merged.

**Comment by ZhuozhaoLi** [comment]
I missed this PR. Is it a prerequisite PR for task status and task info logging? How about those states reported from the endpoint? for example, `waiting-for-nodes`, `waiting-for-launch`, etc.

**Comment by Loonride** [comment]
> I missed this PR. Is it a prerequisite PR for task status and task info logging? How about those states reported from the endpoint? for example, `waiting-for-nodes`, `waiting-for-launch`, etc.

@ZhuozhaoLi Yes, it adds some task status and info logging that we can track. We have only added it in the bare minimum of a couple places in the web service, a couple places on the forwarder, and 1 place in the websocket service. We can add more state info in the future like `waiting-for-nodes`, `waiting-for-launch`, etc.

**Pull request merged**
